src/sentry/api/endpoints/project_plugin_details.py

```diff
@@ -87,6 +90,30 @@ def post(self, request: Request, project, plugin_id) -> Response:
         if not plugin.can_disable:
             return Response({"detail": ERR_ALWAYS_ENABLED}, status=400)
 
+        if integrations.exists(plugin_id):
+            for feature in integrations.get(plugin_id).features:
+                feature_flag_name = "organizations:integrations-%s" % feature.value
+                if not features.has(feature_flag_name, project.organization):
+                    return Response(
+                        {"detail": ERR_FEATURE_REQUIRED % feature_flag_name}, status=403
+                    )
+        else:  # assume it is a plugin
+            for feature in plugin.feature_descriptions:
+                if feature.featureGate == IntegrationFeatures.SESSION_REPLAY:
+                    # NOTE: there is a clash between Sentry native "organizations:session-replay" which is False by default
```

Comment:
@oioki this comment is confusing me, are you saying we have to map `organizations:integrations-session-replay` to `organizations:session-replay`?

Reply:
Currently, `IntegrationFeatures.SESSION_REPLAY` is only used by SessionStack plugin. As this is a plugin (not an integration), its feature flag name is without `integrations-` according to this logic: https://github.com/getsentry/sentry/pull/16825
We can ignore this check because SessionStack is available for all billing plans anyway.

I see that migrating plugins' features to `organizations:plugins-*` might have a point to avoid feature conflicts, but that's a different topic.